### ISSUE TITLE
async/await uses a Promise from core-js that runs outside Fiber

### ISSUE THREAD

**Issue by kamilkisiela** [meteor/meteor#9067]
When JavaScript library uses the `async/await` and compiles it using babel to es5 it's being transformed to code that uses `asyncToGenerator` from `babel-runtime` package.

`babel-runtime/helpers/asyncToGenerator` module uses a `Promise` from the `core-js` package.

Inside of `core-js`:

https://github.com/zloirock/core-js/blob/9bf000955a615307254cdb8861bfdc19c7184d77/library/modules/es6.promise.js#L26-L36

As you can see, `USE_NATIVE` value is Boolean that tells if it should use native `Promise` or just a polyfill.

I did some deep digging and it seems like at the time when an anonymous function executes to return a value for `USE_NATIVE`, there is a `Promise` available and it's wrapped with Fiber which is what we're looking for.

`USE_NATIVE` should be `true` but it's `false`. It happens because inside the `try/catch` an error occurs.

> TypeError: es6PromiseThen.call is not a function

It points to this part of a code from `meteor-promise` package:

https://github.com/meteor/promise/blob/master/promise_server.js#L36

Because `USE_NATIVE === false`, we use a polyfill of `Promise` from `core-js`, This is why everything inside a `async` function runs outside `Fiber`.

## Reproduction

https://github.com/js-accounts/graphql/issues/16#issue-254831880

**Comment by benjamn** [comment]
Thanks for this analysis!

My hunch is that the `TypeError` should be fixable, which would allow `USE_NATIVE` to be true, but I'll need to dig into why that exception is being thrown.

If that doesn't work out, another solution might be to enable `Fiber` wrapping for the `core-js` promise:
```js
require("meteor-promise").enable(
  require("core-js/es6/promise"),
  require("fibers")
);
```

**Comment by kamilkisiela** [comment]
@benjamn If you have some ideas, pointers or anything I could help with, feel free to message me to quicker push it forward.

**Comment by maxfi** [comment]
I'm experiencing the same issue in my app (`METEOR@1.5.2.2`) as well where my own package ([`node-securepay`](https://github.com/maxfi/node-securepay)) overrides meteor's `Promise` polyfill. This happens as  soon as I `import SecurePay from 'node-securepay'`. 

I'm still new to babel but I was under the impression that the `'transform-runtime'` plugin was meant to avoid this by creating ["a sandboxed environment for your code"](https://babeljs.io/docs/plugins/transform-runtime/#why-). However this doesn't appear to be the case as @kamilkisiela mentions. 

I realise that this isn't directly a meteor issue and as the author of the [`node-securepay`](https://github.com/maxfi/node-securepay) package I can make changes to rectify this - _although I have no idea how to accomplish that any advice would be greatly appreciated_! 

So, from the meteor perspective, I assume that many other packages are using `babel`/`core-js` and if these packages override meteor's implementations of certain polyfills then this could be a great source of confusion for meteor developers:

* https://forums.meteor.com/t/promise-await-is-not-a-function/37499/4
* https://forums.meteor.com/t/solved-having-trouble-with-promises-in-meteor-methods/34198/16
* https://forums.meteor.com/t/promises-async-await-in-meteor/33273/11

As such, is it possible to ensure that meteors polyfills cannot be overridden or at least meteor logs a warning with further information for the developer? 

@benjamn how would you recommend working around this on meteor `METEOR@1.5.2.2`? I can't currently upgrade to v1.6 as it breaks other dependencies of my app, specifically `GroundDB`. Are there any other issues that using your above suggested workaround could cause?  Thanks.

**Comment by benjamn** [comment]
I believe this now works in Meteor 1.6.x, by the way. I still don't have any great advice for 1.5.x apps, other than updating to Meteor 1.6 (which is a good idea anyway!).

**Comment by hwillson** [comment]
Closing as this should now be resolved in Meteor 1.6.x.
